common/detector.py

```diff
@@ -13,17 +13,25 @@ def __init__(self, regular_mbox):
         self.inbox = regular_mbox
 
     @abc.abstractmethod
+    def update_sender_profile(self, email):
+        """Updates sender to profile map with a single email.
+
+        Keyword arguments:
+        email -- the email message to add to the sender profile
+        """
+        return
+
     def create_sender_profile(self, num_samples):
         """Creates sender to profile map.
 
         Keyword arguments:
         num_samples -- number of samples to train sender profile on.
 
         Sets self.sender_profile to a dictionary mapping senders to profiles.
-
-        Returns self.sender_profile.
         """
-        return
+        for i in range(num_samples):
```

Comment:
OK.  My worry about adding `self.sender_profile = {}` is that different detectors have different data structures.

Here's what I did instead.  I added extra code to `create_sender_profile()` to detect if it is called twice and raise an exception if it is.  Does that seem reasonable?  (The code is a bit ugly, and apologies for polluting the git history.)